proto/redpanda/core/admin/v2/datalake.proto

```diff
@@ -0,0 +1,83 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package redpanda.core.admin.v2;
+
+import "proto/redpanda/core/pbgen/options.proto";
+import "proto/redpanda/core/pbgen/rpc.proto";
+
+option (redpanda.core.pbgen.cpp_namespace) = "proto::admin";
+
+service DatalakeService {
+    // Gets the coordinator state for the given topics.
+    rpc CoordinatorGetState(CoordinatorGetStateRequest)
+        returns (CoordinatorGetStateResponse) {
+        option (redpanda.core.pbgen.rpc) = {
+            authz: SUPERUSER
+        };
+    }
+}
+
+message CoordinatorGetStateRequest {
+    // Names of topics whose state to return. If empty, returns state for all
+    // topics.
+    repeated string topics_filter = 1;
+}
+
+message CoordinatorGetStateResponse {
+    map<string, TopicState> topic_states = 1;
+}
+
+message DataFile {
+    string remote_path = 1;
+    uint64 row_count = 2;
+    uint64 file_size_bytes = 3;
+    int32 table_schema_id = 4;
+    int32 partition_spec_id = 5;
+    repeated bytes partition_key = 6;
+}
+
+message TranslatedOffsetRange {
+    int64 start_offset = 1;
+    int64 last_offset = 2;
+    repeated DataFile files = 3;
```

Comment:
maybe data_files to more clearly separate it from dlq_files?

Reply:
Done